src/main/java/ui/components/pickers/BoardPickerState.java

```diff
@@ -0,0 +1,77 @@
+package ui.components.pickers;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Liu Xinan
+ */
+class BoardPickerState {
+
+    private Set<String> boards;
+    private List<String> matchedBoards;
+    private Optional<String> suggestion;
+    private String keyword;
+
+    BoardPickerState(Set<String> boards, String userInput) {
+        this(boards, new ArrayList<>(), userInput.trim(), Optional.empty());
+        update();
+    }
+
+    private BoardPickerState(Set<String> boards, List<String> matchedBoards, String keyword,
+                             Optional<String> suggestion) {
+        this.boards = boards;
+        this.matchedBoards = matchedBoards;
+        this.suggestion = suggestion;
+        this.keyword = keyword;
+    }
+
+    List<String> getMatchedBoards() {
+        return matchedBoards;
+    }
+
+    Optional<String> getSuggestion() {
+        return suggestion;
+    }
+
+    private void update() {
+        if (keyword.isEmpty()) {
+            matchAllBoards();
+            return;
+        }
+
+        updateMatchedBoards();
+        updateSuggestion();
+    }
+
+    private void matchAllBoards() {
+        matchedBoards = boards.stream().collect(Collectors.toList());
+    }
+
+    private void updateMatchedBoards() {
+        String[] prefixes = keyword.split("\\s+");
```

Comment:
So there should be multiple `keywords`? If that's the case then the variable should be `keywords` instead of `keyword`